plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -17,25 +17,43 @@
 import org.testcontainers.utility.DockerImageName;
 
 import java.sql.Connection;
+import java.sql.SQLException;
 import java.sql.Statement;
 
+import static java.util.Objects.requireNonNull;
+
 public final class TestingSqlServer
         extends MSSQLServerContainer<TestingSqlServer>
 {
+    private String catalogName = "master";
+
     public TestingSqlServer()
     {
         super(DockerImageName.parse("microsoft/mssql-server-linux:2017-CU13").asCompatibleSubstituteFor("mcr.microsoft.com/mssql/server:2017-CU12"));
         addEnv("ACCEPT_EULA", "yes");
     }
 
+    public void setCurrentCatalog(String catalogName)
```

Comment:
The class was immutable until now. 

if the class needs to know the catalog, we should pass the catalog name (and desired snapshot isolation) in the ctor and let this class create them